src/Services/AuthenticationService.php

```diff
@@ -190,4 +273,61 @@ protected function addRelyingPartyAssociation(ClientEntityInterface $oidcClient,
             ),
         );
     }
+
+    /**
+     * This is a wrapper around Auth/State::loadState that facilitates testing by
+     * hiding the static method
+     *
+     * @param   array  $queryParameters
+     *
+     * @return array
+     * @throws NoState
+     */
+    public function loadState(array $queryParameters): array
+    {
+        if (empty($queryParameters[ProcessingChain::AUTHPARAM])) {
+            throw new NoState();
+        }
+        $stateId = (string)$queryParameters[ProcessingChain::AUTHPARAM];
+        return State::loadState($stateId, ProcessingChain::COMPLETED_STAGE);
+    }
+
+    /**
+     * Run authproc filters with the processing chain
+     * Creating the ProcessingChain require metadata.
+     * - For the idp metadata use the OIDC issuer as the entityId (and the authprocs from the main config file)
+     * - For the sp metadata use the client id as the entityId (and don’t set authprocs).
+     *
+     * @param   array  $state
+     *
+     * @return void
+     * @throws Exception
+     * @throws Error\UnserializableException
+     */
+    protected function runAuthProcs(array &$state): void
+    {
+        $idpMetadata = [
+            'entityid' => $state['Source']['entityid'] ?? '',
+            // ProcessChain needs to know the list of authproc filters we defined in module_oidc configuration
+            'authproc' => $this->moduleConfig->getAuthProcFilters(),
+        ];
+        $spMetadata = [
+            'entityid' => $state['Destination']['entityid'] ?? '',
+        ];
+        $pc = new ProcessingChain(
+            $idpMetadata,
+            $spMetadata,
+            explode('.', (string)$this->moduleConfig::OPTION_AUTH_PROCESSING_FILTERS)[1],
```

Comment:
I think this is more readable if we remove the whole `explode` bit and just use the string `oidc`

Reply:
done